packages/notte-core/src/notte_core/utils/raw_file.py

```diff
@@ -0,0 +1,54 @@
+import mimetypes
+import re
+import time
+from typing import Any
+from urllib.parse import urlparse
+
+import requests
+
+from notte_core.browser.dom_tree import ComputedDomAttributes, DomAttributes, DomNode, NodeSelectors
+from notte_core.browser.node_type import NodeRole, NodeType
+
+
+def get_file_ext(headers: dict[str, Any]) -> str:
+    return mimetypes.guess_extension(headers["content-type"]) or ""
+
+
+def get_filename(headers: dict[str, Any], url: str) -> str:
```

Comment:
can you add some tests also for this function ? With different headers type so that every if/else case is covered